gordo_components/dataset/datasets.py

```diff
@@ -86,42 +72,150 @@ def __init__(
             )
 
     def get_data(self) -> Tuple[pd.DataFrame, Optional[pd.DataFrame]]:
-
         series_iter: Iterable[pd.Series] = self.data_provider.load_series(
             from_ts=self.from_ts,
             to_ts=self.to_ts,
             tag_list=list(set(self.tag_list + self.target_tag_list)),
         )
-        data: pd.DataFrame = self.join_timeseries(
-            series_iter,
-            self.from_ts,
-            self.to_ts,
-            self.resolution,
-            aggregation_methods=self.aggregation_methods,
-        )
+        data = pd.concat(series_iter, axis=1, join="inner")
         if self.row_filter:
             data = pandas_filter_rows(data, self.row_filter)
+        return self._X_y_from_dataframe(data)
+
+    def _X_y_from_dataframe(
+        self, data: pd.DataFrame
+    ) -> Tuple[pd.DataFrame, Optional[pd.DataFrame]]:
+        """
+        From a dataframe, which is a assumed to have columns which are a superset
+        of ``tag_list`` and ``target_tag_list``, return ``X`` and ``y``
 
+        Parameters
+        ----------
+        data: pd.DataFrame
+            DataFrame expected to have all target and training tags as columns
+
+        Returns
+        -------
+        Tuple[pd.DataFrame, Optional[pd.DataFrame]]:
+        """
         x_tag_names = [tag.name for tag in self.tag_list]
         y_tag_names = [tag.name for tag in self.target_tag_list]
 
         X = data[x_tag_names]
         y = data[y_tag_names] if self.target_tag_list else None
-
         return X, y
 
     def get_metadata(self):
         metadata = {
             "tag_list": self.tag_list,
             "target_tag_list": self.target_tag_list,
-            "train_start_date": self.from_ts,
-            "train_end_date": self.to_ts,
-            "resolution": self.resolution,
+            "train_start_date": self.from_ts.isoformat(),
+            "train_end_date": self.to_ts.isoformat(),
             "filter": self.row_filter,
         }
         return metadata
 
 
+class TimeSeriesDataset(RawTimeSeriesDataset):
```

Comment:
It is a bit subpar that important arguments to TimeSeriesDataset are no longer documented directly in its init, but one has to chase them down in the superclass.  Also, they are required, but that is no longer visible from the signature :-s 
IDK if this is a argument for adding them here as explicit arguments, or for merging the two raw and non-raw timeseriesdatasets